fuse_core/include/fuse_core/throttled_callback.h

```diff
@@ -52,28 +56,32 @@ template <class Callback>
 class ThrottledCallback
 {
 public:
-  // TODO(CH3): Keep the use of "use_wall_time", but pass in a clock as appropriate to use
-  // non-wall-time.
-  //
-  // TODO(CH3): Add getters into any class that could add ThrottledCallbacks to obtain their
-  // NodeClockInterface to then get the clock to pass into this!
   /**
    * @brief Constructor
    *
    * @param[in] keep_callback   The callback to call when kept, i.e. not dropped. Defaults to nullptr
    * @param[in] drop_callback   The callback to call when dropped because of the throttling. Defaults to nullptr
    * @param[in] throttle_period The throttling period duration in seconds. Defaults to 0.0, i.e. no throttling
    * @param[in] use_wall_time   Whether to use wall time or not. Defaults to false
+   * @throws if the use_wall_time is false and no NodeInterfaces is passed in
    */
   ThrottledCallback(Callback&& keep_callback = nullptr,  // NOLINT(whitespace/operators)
                     Callback&& drop_callback = nullptr,  // NOLINT(whitespace/operators)
                     const rclcpp::Duration& throttle_period = rclcpp::Duration(0,0),
-                    const bool use_wall_time = false)
+                    const bool use_wall_time = false,
+                    node_interfaces::NodeInterfaces<
+                      node_interfaces::Clock
+                    >::SharedPtr interfaces = nullptr)
```

Comment:
Looking at the implementation, the `use_wall_time` and `interfaces` arguments seem indirect. The `ThrottledCallback` class needs a persistent clock; I don't think it needs to concern itself with where the clock comes from. I would recommend replacing both of these arguments with a single argument `rclcpp::Clock::SharedPtr clock`.

That both simplifies the implementation, and would allow things like passing in a steady time clock.

Reply:
https://github.com/locusrobotics/fuse/pull/286/commits/d9493d49276c9dcab014a65aed5ffbf15aa7d714

Signature changed! I am still keeping usage of `use_wall_time` on the caller side though, at which point an appropriate clock will be set.